sixtyfps_compiler/generator/rust.rs

```diff
@@ -1398,11 +1398,12 @@ fn compile_expression(expr: &Expression, ctx: &EvaluationContext) -> TokenStream
             debug_assert!(matches!(array.ty(ctx), Type::Array(_)));
             let base_e = compile_expression(array, ctx);
             let index_e = compile_expression(index, ctx);
-            quote!(match &#base_e { x => {
+            quote!({
+                let x = &(#base_e);
```

Comment:
that's not the same.  We want to make sure that the temporaries in #base_e are not destroyed.

Reply:
I do not get what you are going at, sorry.

Reply:
```rust
let x = &String::new("42").trim();   // doesn't work (take address of temporary)
println!("{}", x); 
```
```rust
match (&String::new("42").trim()) {
  // works
  x => { println!("{}", x); }
```

So please undo that.

Why are you changing it?

Reply:
I guess something like this would fail to compile in rust:

```
property <string> world: (["hello", "world"])[1];
```

Reply:
Wow, tricky, but obvious now that you pointed it out!